### PR TITLE
[cmake] Avoid warning when building from tar file

### PR DESCRIPTION
**Pull request by bellenot**
Use the ROOT version from file when building from the source tar file
Fixes https://github.com/root-project/root/issues/15178

### CODE REVIEW

File: cmake/modules/SetROOTVersion.cmake [hunk 1]
```diff
@@ -82,7 +82,9 @@ function(SET_ROOT_VERSION)
     if (${GIT_DESCRIBE_ALL} MATCHES "^tags/")
       string(REGEX REPLACE "^tags/" "" GIT_DESCRIBE_ALWAYS ${GIT_DESCRIBE_ALL})
     else()
-      message(WARNING "GIT_DESCRIBE_ALL is `${GIT_DESCRIBE_ALL}` with a format unexpected for tags. Not setting the commit.")
+      set(ROOT_GIT_VERSION "${ROOT_MAJOR_VERSION}-${ROOT_MINOR_VERSION}-${ROOT_PATCH_VERSION}")
```
Comment (pcanal) [inline cmake/modules/SetROOTVersion.cmake:85]: Should we check that those variables are set before using them? (and if they are not still issue the warning?)

Comment (bellenot) [reply cmake/modules/SetROOTVersion.cmake:85]: Well, the complain from Atlas is about the warning, not about the variables not being set...

Comment (bellenot) [reply cmake/modules/SetROOTVersion.cmake:85]: And I just found more elegant to artificially set the variables using `RVersion.h` than letting them empty...

**Review by dpiparo — APPROVED** [review]
LGTM.

File: cmake/modules/SetROOTVersion.cmake (hunk 1 above)
Comment (pcanal) [reply cmake/modules/SetROOTVersion.cmake:85]: That sounds good.  Do I understand correctly that `ROOT_MAJOR_VERSION`, `ROOT_MINOR_VERSION` and `ROOT_PATCH_VERSION` are always guaranteed to be set and we will never ends with `ROOT_GIT_VERSION == "--"`?

Comment (bellenot) [reply cmake/modules/SetROOTVersion.cmake:85]: They are taken from `${CMAKE_SOURCE_DIR}/core/foundation/inc/ROOT/RVersion.hxx`. So if they are not guaranteed to be set then we are in big trouble I think...

Comment (pcanal) [reply cmake/modules/SetROOTVersion.cmake:85]: Perfect.  Thanks for the clarification.

**Review by pcanal — APPROVED** [review]

**Pull request merged**